tests/Integration/Behaviour/Features/Context/Domain/ApiAccessManagementFeatureContext.php

```diff
@@ -0,0 +1,156 @@
+use PrestaShop\PrestaShop\Core\Domain\ApiAccess\Command\AddApiAccessCommand;
+use PrestaShop\PrestaShop\Core\Domain\ApiAccess\Command\EditApiAccessCommand;
+use PrestaShop\PrestaShop\Core\Domain\ApiAccess\Exception\ApiAccessConstraintException;
+use PrestaShop\PrestaShop\Core\Domain\ApiAccess\Query\GetApiAccessForEditing;
+use PrestaShop\PrestaShop\Core\Domain\ApiAccess\ValueObject\ApiAccessId;
+use Tests\Integration\Behaviour\Features\Context\CommonFeatureContext;
+use Tests\Integration\Behaviour\Features\Context\Util\DataComparator;
+use Tests\Integration\Behaviour\Features\Context\Util\DataTransfer;
+use Tests\Resources\Resetter\ApiAccessResetter;
+
+class ApiAccessManagementFeatureContext extends AbstractDomainFeatureContext
+{
+    /**
+     * "When" steps perform actions, and some of them store the latest result
+     * in this variable so that "Then" action can check its content
+     */
+    protected mixed $latestResult;
+
+    /**
+     * Registry to keep track of created/edited api access using references
+     *
+     * @var int[]
+     */
+    protected array $apiAccessRegistry = [];
+
+    /**
+     * @When /^I create an api-access "(.+)" with following properties:$/
+     */
+    public function createApiAccessUsingCommand($apiAccessReference, TableNode $table)
+    {
+        $data = $table->getRowsHash();
+        $commandBus = $this->getCommandBus();
+
+        $command = new AddApiAccessCommand(
+          $data['clientName'],
+          $data['apiClientId'],
+          (bool) $data['enabled'],
+          $data['description']
+        );
+
+        try {
+            /** @var ApiAccessId $id */
+            $id = $commandBus->handle($command);
+
+            $this->latestResult = $id->getValue();
+            $this->apiAccessRegistry[$apiAccessReference] = $id->getValue();
+        } catch (ApiAccessConstraintException $e) {
+            $this->setLastException($e);
+        }
+    }
+
+    /**
+     * @When /^I query api-access "(.+)" I should get a ApiAccess with properties:$/
+     */
+    public function assertQueryApiAccessProperties($apiAccessReference, TableNode $table)
+    {
+        $expectedData = $table->getRowsHash();
+        $this->assertApiAccessReferenceExistsInRegistry($apiAccessReference);
+
+        $queryBus = $this->getQueryBus();
+        $result = $queryBus->handle(new GetApiAccessForEditing($this->apiAccessRegistry[$apiAccessReference]));
+
+        $serializer = CommonFeatureContext::getContainer()->get('serializer');
+        $realData = $serializer->normalize($result);
+
+        DataComparator::assertDataSetsAreIdentical($expectedData, $realData);
+
+        $this->latestResult = null;
+    }
+
+    /**
+     * @When /^I edit api-access "(.+)" and I change the following properties:$/
+     */
+    public function editCustomerUsingCommand($apiAccessReference, TableNode $table)
+    {
+        $this->assertApiAccessReferenceExistsInRegistry($apiAccessReference);
```

Comment:
```suggestion
```
The shared storage will throw an exception when we access unknown references so no need for this check